src/heap.rs

```diff
@@ -20,6 +20,30 @@ impl HeapId {
 #[derive(Debug, Clone, PartialEq)]
 pub struct Heap(Vec<u8>);
 
+impl Heap {
+    pub fn resize(&mut self, new_len: usize, value: u8) {
```

Comment:
Do we actually need resize and len?

I would like to keep the backing storage as an implementation detail if possible.